source/Projectile.cpp

```diff
@@ -71,18 +72,20 @@ Projectile::Projectile(const Projectile &parent, const Point &offset, const Angl
 	cachedTarget = TargetPtr().get();
 	double inaccuracy = weapon->Inaccuracy();
 	if(inaccuracy)
-	{
 		this->angle += Angle::Random(inaccuracy) - Angle::Random(inaccuracy);
-		if(!parent.weapon->Acceleration())
-		{
-			// Move in this new direction at the same velocity.
-			// To maintain the sign of the velocity, Point::Length can’t be used.
-			Point referenceVector = parent.angle.Unit();
-			double parentVelocity = referenceVector.Dot(parent.velocity);
-			velocity = this->angle.Unit() * parentVelocity;
-		}
-	}
-	velocity += this->angle.Unit() * (weapon->Velocity() + Random::Real() * weapon->RandomVelocity());
+
+	// Given that submunitions inherit the velocity of the parent projectile,
+	// it is often the case that submunitions don't add any additional velocity.
+	// But we still want inaccuracy to have an effect on submunitions. Because of
+	// this, we tilt the velocity of submunitions in the direction of the inaccuracy.
+
+	// Calculate the speed of this projectile, which is the speed of the parent
+	// plus whatever speed the submunition adds.
+	speed = parent.speed + weapon->Velocity() + Random::Real() * weapon->RandomVelocity();
+	// Unwind the inaccuracy of the parent.
+	velocity -= parent.angle.Unit() * parent.speed;
+	// Apply the inaccuracy of the submunition.
+	velocity += this->angle.Unit() * speed;
```

Comment:
```suggestion
	// Calculate the speed of this projectile, which is the speed of the parent
	// plus whatever speed the submunition adds.
	speed = parent.speed + weapon->Velocity() + Random::Real() * weapon->RandomVelocity();

	// Given that submunitions inherit the velocity of the parent projectile,
	// it is often the case that submunitions don't add any additional velocity.
	// But we still want inaccuracy to have an effect on submunitions. Because of
	// this, we tilt the velocity of submunitions in the direction of the inaccuracy.
	// This isn't done for submunitions with an accelerating parent, as an accelerating
	// projectile's actual velocity won't match the recorded speed value.
	if(inaccuracy && !parent.weapon->Acceleration())
	{
		// Unwind the inaccuracy of the parent.
		velocity -= parent.angle.Unit() * parent.speed;
		// Apply the inaccuracy of the submunition.
		velocity += this->angle.Unit() * speed;
	}
	else
		velocity += this->angle.Unit() * (speed - parent.speed);
```

Reply:
I'll look at this when I can later on.  It's miss

Reply:
I'm going to commit this as a starter but we're running into the same issue of multi-stage weapons, if they have acceleration.  And it should be worse for multi-stage weapons that may swap between acceleration and unguided flight.  My first intuition would be to change `speed` to only be updated by munitions without acceleration.

Reply:
I think what you're saying is that a single layer of/submunition with acceleration throws off everything that came before it.  Your logic should be fine as is then, but I am going to pursue the `inaccuracyVelocity` I mentioned above.